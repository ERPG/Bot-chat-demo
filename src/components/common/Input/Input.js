import React from 'react';

import './Input.css';

/**
 * Represents a base Input.
 * @constructor
 * @param {string} value - Input value.
 * @param {React.ChangeEvent<HTMLInputElement>} onChange - Action to execute on input change.
 * @param {React.KeyboardEvent<HTMLInputElement>} onKeyUp - Action to execute on keyboard event.
 * @param {string} placeholder - input placeholder text.
 * There are more props in this element, for the purpose of this test we will use this few.
 */

const Input = ({value, onChange, onKeyUp, placeholder }) => {
  return (
    <input
      data-landbot='chat-input'
      className="landbot-input"
      type="text"
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      onKeyUp={onKeyUp}
    />
  )
}

export default Input;