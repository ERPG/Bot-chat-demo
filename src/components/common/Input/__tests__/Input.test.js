import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { customRender } from '../../../../../__tests__/__utils__/testUtils';
import Input from '../Input';

describe('Input component', () => {
  const inputSelector = 'chat-input';
  const inputProps = {
    value: '',
    onChange: jest.fn(),
    onKeyUp: jest.fn(),
    placeholder: 'placeholder'
  }
  beforeEach(() => {
    jest.clearAllMocks();
  })

  it('Should render component without errors', () => {
    const { container } = render(<Input {...inputProps}/>);
    expect(container).toMatchSnapshot();
  });

  it('Should call onChange event', async() => {
    const { getByDataLb } = customRender(<Input {...inputProps} />);
    const input = getByDataLb(inputSelector)
    expect(input).toBeInTheDocument();

    fireEvent.change(input, {target: {value: 'landbot'}})

    expect(inputProps.onChange).toHaveBeenCalled();
  });

  it('Should have a placeholder', () => {
    const { getByDataLb } = customRender(<Input {...inputProps} />);
    const input = getByDataLb(inputSelector)
    expect(input).toBeInTheDocument();

    expect(input.placeholder).toBe('placeholder');
  });
});