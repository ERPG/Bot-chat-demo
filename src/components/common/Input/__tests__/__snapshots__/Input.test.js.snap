// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should render component without errors 1`] = `
<div>
  <input
    class="landbot-input"
    data-landbot="chat-input"
    placeholder="placeholder"
    type="text"
    value=""
  />
</div>
`;
